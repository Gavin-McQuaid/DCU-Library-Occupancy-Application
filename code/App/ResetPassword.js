import React, {Component} from 'react';
import { Image, KeyboardAvoidingView, View, TextInput, Alert,StyleSheet, TouchableOpacity, Text, StatusBar} from 'react-native';
import {StackNavigator} from 'react-navigation';
import { NavigationActions } from 'react-navigation';
import {hook} from 'cavy';

class ResetPassword extends Component{
	static navigationOptions = {
		title : 'ResetPassword'
	}

	constructor(props){
		super(props)
		this.state = {
			NewPassword: '',
			NewPasswordConfirm: '',
			PasswordError: '',
			PasswordConfirmationError: ''		
		}
	}

	// Function that checks entered password is at least eight characters long
	ValidatePassword = () => {	
	    if(this.state.NewPassword.length < 8){
			this.setState({PasswordError: "Passwords must not be empty and at least 8 characters long"})
			return false
		}
		else{
			this.setState({PasswordError: ""})
			return true
		}
	}

	// Function that checks entered password matches password confirmation field
	PasswordsMatch = () => {
		if(this.state.NewPassword !== this.state.NewPasswordConfirm || this.state.NewPasswordConfirm.length < 8){
			this.setState({PasswordConfirmationError: "Passwords don't match"})
			return false
		}
		else{
			this.setState({PasswordConfirmationError: ""})
			return true
		}
	}
	
   // Carries out one final validation check before password can be reset
   FinalValidation = () => {
		passwordOk = this.ValidatePassword();
		passwordsMatch = this.PasswordsMatch();
		if(passwordOk){
			this.ResetPassword();}
		}


	// Function that resets a user's password to the new password they've entered and confirmed
	ResetPassword = () => {
		const {NewPassword} = this.state;
		const {NewPasswordConfirm} = this.state;
		const resetAction = NavigationActions.reset({
  index: 0,
  actions: [NavigationActions.navigate({ routeName: 'Login'})],
});
		fetch('http://ec2-34-244-240-250.eu-west-1.compute.amazonaws.com/ResetPassword.php', {
			method: 'POST',
			headers: {
				'Accept' : 'application/json',
				'Content-Type' : 'applcation/json'
			},
			body: JSON.stringify({
				password: NewPassword,
				password_confirmation: NewPasswordConfirm,
				email: this.props.navigation.state.params.email
			}) 
		}).then((response) => response.json())
      .then((responseJson) => {

		if(responseJson == "Success"){
			Alert.alert("Password reset successfully");
			this.props.navigation.dispatch(resetAction);
		}
      }).catch((error) => {
        console.error(error);
      });
       }

	render() {
		return (
		<KeyboardAvoidingView behavior = 'padding' style = {styles.container}>
			<View style = {styles.container}>
				<StatusBar
					barStyle = 'light-content'
				/>	
				<View style = {styles.logoContainer}>
					<Text style = {styles.title}> DCU Library App 
					</Text>
					<Image
						style = {styles.logo}
						source = {require('./images/logo.png')} 
					/>
				</View>
				<TextInput
					ref = {this.props.generateTestHook('ResetPassword.PasswordTextInput')}
					placeholder = "Password"
					placeholderTextColor = 'white'
					returnKeyType = 'next'
					autoCapitalize = 'none'
					secureTextEntry
					style = {styles.input}
					onChangeText ={NewPassword => this.setState({NewPassword},() => this.ValidatePassword())}
				/>
				<Text style = {styles.errorText} >{this.state.PasswordError}
				</Text>
				<TextInput
					ref = {this.props.generateTestHook('ResetPassword.ConfirmPasswordTextInput')}
					placeholder = "Confirm password"
					placeholderTextColor = 'white'
					returnKeyType = 'go'
					autoCapitalize = 'none'
					secureTextEntry
					style = {styles.input}
					onChangeText ={NewPasswordConfirm=> this.setState({NewPasswordConfirm}, () => this.PasswordsMatch())}
				/>
				<Text style = {styles.errorText} >{this.state.PasswordConfirmationError}
				</Text>
				<TouchableOpacity
					ref = {this.props.generateTestHook('ResetPassword.ResetButton')}
					style = {styles.buttonContainer}
					onPress={() => this.FinalValidation()}>
					<Text style = {styles.buttontext}> Reset Password
					</Text>
				</TouchableOpacity>		
			</View>
		</KeyboardAvoidingView>	
		)
	}
}
export default hook(ResetPassword);
const styles = StyleSheet.create({
  container: {
	backgroundColor : 'cornflowerblue',  
    flex: 1,

  },

  logoContainer : {
	alignItems : 'center',
	flexGrow : 1,
	justifyContent : 'center'
  
  },

  logo: {
	width : 200,
	height : 150
  },
  title: {
	color : 'white',
	marginBottom : 10,
	fontSize : 30,
	opacity : 0.9,
	justifyContent: 'center'
	
  },

  errorText:{
  	color: 'white'
  },

  input: { 
	height: 40,
	backgroundColor : 'rgba(255,255,255,0.2)',
	marginBottom : 15,
	color : 'white',
	paddingHorizontal : 10,
	borderRadius : 25
	},

  buttonContainer : {
	backgroundColor :'steelblue',
	paddingVertical : 15,
	marginBottom : 15,
	borderRadius : 25
	},

  buttontext : {
	color : 'white',
	textAlign : 'center',
	fontWeight : '400'
	}

});	