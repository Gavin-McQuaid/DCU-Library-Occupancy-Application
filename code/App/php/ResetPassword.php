<?php include '../inc/dbinfo.inc';

  /* Connect to MySQL and select the database. */
  $connection = mysqli_connect(DB_SERVER, DB_USERNAME, DB_PASSWORD);
  if (mysqli_connect_errno()) echo 'Failed to connect to MySQL: ' . mysqli_connect_error();
  $database = mysqli_select_db($connection, DB_DATABASE);
  $json = file_get_contents('php://input');
  $jsonobject = json_decode($json,true);
  $email = $jsonobject['email'];
  $newPassword = md5($jsonobject['password']);
  $newPasswordConfirmation = md5($jsonobject['password_confirmation']);
 if(strlen($newPassword) >= 8 && $newPassword == $newPasswordConfirmation){

  $query = "UPDATE users SET password = '$newPassword' WHERE email = '$email'";
  if(mysqli_query($connection, $query)){
        $resetSuccessMsg = 'Success';
        $jsonEncodeMsg = json_encode($resetSuccessMsg);
        echo $jsonEncodeMsg;

}
else{
$resetFailureMsg = 'Failed to reset password';
$jsonEncodeFailure = json_encode($resetFailureMsg);
echo $jsonEncodeFailure;
}}
mysqli_close($connection);
?>
