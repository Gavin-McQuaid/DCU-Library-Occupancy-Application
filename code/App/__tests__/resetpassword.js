import React from 'react';
import ResetPassword from '../ResetPassword';
import renderer from 'react-test-renderer';
import {shallow, mount,render,configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import mockStore from 'redux-mock-store';
const Text = require('Text');
const TouchableHighlight = require('TouchableHighlight');
const ReactTestRenderer = require('react-test-renderer');
const TouchableOpacity = require('TouchableOpacity');
const TextInput = require('TextInput');
const Image = require('Image');

var ReactTestUtils = require('react-dom/test-utils')
configure({ adapter: new Adapter() })

describe('ResetPassword', () =>{
	it('text shows', () =>{
		const instance0 = ReactTestRenderer.create(
			<Text style = {{}}> DCU Library App </Text>	
		);
		});
	it('image shows', () =>{
		const instance1 = ReactTestRenderer.create(
			<Image style = {{}} source = {require('./Images/logo.png')}> </Image>
		);
		
	});
	it('TextInput 1', () => {
		const instance2 = ReactTestRenderer.create(
		<TextInput placeholder = 'Password'></TextInput>
		);	
	});
	it('TextInput 2', () => {
		const instance2 = ReactTestRenderer.create(
		<TextInput placeholder = 'Confirm Password'></TextInput>
		);	
	});
		it('touchable opacity shows', () =>{
		const instance3 = ReactTestRenderer.create(
			<TouchableOpacity style ={{}}>
			<Text> Reset Password </Text>
			</TouchableOpacity>
		);
	});
	it ('Text Inputs works', () => {
		const wrapper = shallow(<ResetPassword />);
		const TextInput1= wrapper.find('TextInput').at(0);
		const TextInput2= wrapper.find('TextInput').at(1);
		const credentials = { password : 'hahaha', confirmpassword : 'hahaha'};
		TextInput1.value = credentials.password;
		expect(TextInput1.value).toBe('hahaha');
		TextInput2.value = credentials.confirmpassword;
		expect(TextInput2.value).toBe('hahaha');
	});
	it('touchableOpacity 1 touch works', () => {
		const wrapper = shallow(<ResetPassword />);
		const touchOpacity1 = wrapper.find('TouchableOpacity').at(0);
		const store = mockStore({});
		const navigation = {navigate : jest.fn()};
		expect(renderer.create(<ResetPassword store = {store} navigation = {navigation} />)).toMatchSnapshot();
	});
	it ('Check Reset Password works', () => {
		const wrapper = shallow(<ResetPassword />);
		const TextInput1= wrapper.find('TextInput').at(0);
		const TextInput2= wrapper.find('TextInput').at(1);
		const credentials = { password : 'hahaha', confirmpassword : 'hahaha'};
		TextInput1.value = credentials.password;
		expect(TextInput1.value).toBe('hahaha');
		TextInput2.value = credentials.confirmpassword;
		expect(TextInput2.value).toBe('hahaha');
		const touchOpacity1 = wrapper.find('TouchableOpacity').at(0);
		const store = mockStore({});
		const navigation = {navigate : jest.fn()};
		expect(renderer.create(<ResetPassword store = {store} navigation = {navigation} />)).toMatchSnapshot();
	});
	const rendered = renderer.create(<ResetPassword/>).toJSON();
		expect(rendered).toMatchSnapshot();
	
});