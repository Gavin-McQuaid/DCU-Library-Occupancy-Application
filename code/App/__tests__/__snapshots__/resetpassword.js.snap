// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<View
  keyboardVerticalOffset={0}
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "cornflowerblue",
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "cornflowerblue",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexGrow": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 30,
            "justifyContent": "center",
            "marginBottom": 10,
            "opacity": 0.9,
          }
        }
      >
         DCU Library App
      </Text>
      <Image
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "height": 150,
            "width": 200,
          }
        }
      />
    </View>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="white"
      returnKeyType="next"
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "rgba(255,255,255,0.2)",
          "borderRadius": 25,
          "color": "white",
          "height": 40,
          "marginBottom": 15,
          "paddingHorizontal": 10,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }
    >
      
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="Confirm password"
      placeholderTextColor="white"
      returnKeyType="go"
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "rgba(255,255,255,0.2)",
          "borderRadius": 25,
          "color": "white",
          "height": 40,
          "marginBottom": 15,
          "paddingHorizontal": 10,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }
    >
      
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "steelblue",
          "borderRadius": 25,
          "marginBottom": 15,
          "opacity": 1,
          "paddingVertical": 15,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontWeight": "400",
            "textAlign": "center",
          }
        }
      >
         Reset Password
      </Text>
    </View>
  </View>
</View>
`;

exports[`ResetPassword Check Reset Password works 1`] = `
<View
  keyboardVerticalOffset={0}
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "cornflowerblue",
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "cornflowerblue",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexGrow": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 30,
            "justifyContent": "center",
            "marginBottom": 10,
            "opacity": 0.9,
          }
        }
      >
         DCU Library App
      </Text>
      <Image
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "height": 150,
            "width": 200,
          }
        }
      />
    </View>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="white"
      returnKeyType="next"
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "rgba(255,255,255,0.2)",
          "borderRadius": 25,
          "color": "white",
          "height": 40,
          "marginBottom": 15,
          "paddingHorizontal": 10,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }
    >
      
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="Confirm password"
      placeholderTextColor="white"
      returnKeyType="go"
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "rgba(255,255,255,0.2)",
          "borderRadius": 25,
          "color": "white",
          "height": 40,
          "marginBottom": 15,
          "paddingHorizontal": 10,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }
    >
      
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "steelblue",
          "borderRadius": 25,
          "marginBottom": 15,
          "opacity": 1,
          "paddingVertical": 15,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontWeight": "400",
            "textAlign": "center",
          }
        }
      >
         Reset Password
      </Text>
    </View>
  </View>
</View>
`;

exports[`ResetPassword touchableOpacity 1 touch works 1`] = `
<View
  keyboardVerticalOffset={0}
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "cornflowerblue",
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "cornflowerblue",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexGrow": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 30,
            "justifyContent": "center",
            "marginBottom": 10,
            "opacity": 0.9,
          }
        }
      >
         DCU Library App
      </Text>
      <Image
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "height": 150,
            "width": 200,
          }
        }
      />
    </View>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="white"
      returnKeyType="next"
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "rgba(255,255,255,0.2)",
          "borderRadius": 25,
          "color": "white",
          "height": 40,
          "marginBottom": 15,
          "paddingHorizontal": 10,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }
    >
      
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="Confirm password"
      placeholderTextColor="white"
      returnKeyType="go"
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "rgba(255,255,255,0.2)",
          "borderRadius": 25,
          "color": "white",
          "height": 40,
          "marginBottom": 15,
          "paddingHorizontal": 10,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }
    >
      
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "steelblue",
          "borderRadius": 25,
          "marginBottom": 15,
          "opacity": 1,
          "paddingVertical": 15,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontWeight": "400",
            "textAlign": "center",
          }
        }
      >
         Reset Password
      </Text>
    </View>
  </View>
</View>
`;
